trie/transition.go

```diff
@@ -20,18 +20,21 @@ import (
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/rlp"
 	"github.com/gballet/go-verkle"
 )
 
 type TransitionTrie struct {
 	overlay *VerkleTrie
 	base    *SecureTrie
+	storage bool
```

Comment:
The calling code differentiates between account tries and storage tries. Even though the verkle tree is a single tree, there are multiple transition trees, one for the (unique overlay tree, unique account trie) combination, and one for each of the (unique overlay tree, storage trie for a given account) combination. 

As a result, this flag is added to specify if the current `TransitionTree` corresponds to the account trie or any of the storage tries.

One thing that this flag should be used for, is to not `Commit` when set, otherwise the tree will be committed over and over again, every time an account's storage is updated. This will kill performance, so `Commit` should be a no-op unless `storage == false`.

Reply:
Two questions/points:
- `storage` is only used in the `NewTransitionTree`, so technically "not used". Maybe your comment was telling something that we still have to do? (Just checking we're aware that for now is something unused)
- Your comment made me realize that in `state_processor.go` we never re-set `tt` each time the MPT-tree changes. For example, `state_processor.go` L158 is using `mpt.Hash()` and `mpt` was only assigned once doing `tt.Base()`.. but at that point we might already jumped to another tree, right? Feels like we should reopen `tt` when we jump to another account?

Reply:
* you are correct `storage` isn't used, I should remove it
 * the MPT tree becomes read only, so we shouldn't have to update it. I'm probably missing your point here.

Reply:
Reg the second point, dismiss it. I've been trying to double-check (since I said this three weeks ago :laughing: ) if I was referring to something else, but doesn't seem convincing to me. So I think this is fine!